tests/test_tree.py

```diff
@@ -0,0 +1,176 @@
+from dash import Dash, html, Output, Input, _dash_renderer
+import dash_mantine_components as dmc
+
+_dash_renderer._set_react_version("18.2.0")
+
+
+def tree_app(**kwargs):
+    app = Dash(__name__)
+
+    default_data = [
+        {
+            "label": "1",
+            "value": "1",
+            "children": [
+                {
+                    "label": "a",
+                    "value": "1.a",
+                    "children": [
+                        {"label": "i", "value": "1.a.i"},
+                        {"label": "ii", "value": "1.a.ii"},
+                        {"label": "iii", "value": "1.a.iii"},
+                    ],
+                },
+                {"label": "b", "value": "1.b"},
+            ],
+        },
+        {"label": "2", "value": "2"},
+    ]
+
+    app.layout = dmc.MantineProvider(
+        html.Div(
+            [
+                dmc.Tree(id="tree", data=kwargs.pop("data", default_data), **kwargs),
+                html.Div(id="output-checked"),
+                html.Div(id="output-data"),
+                html.Div(id="output-expanded"),
+                html.Div(id="output-selected"),
+                html.Button(id="button-expand-all", children="expand all"),
+                html.Button(id="button-clear-data", children="clear data"),
+                html.Button(id="button-clear-selection", children="clear selection"),
+                html.Button(id="button-clear-checked", children="clear checked"),
+            ]
+        )
+    )
+
+    @app.callback(Output("output-checked", "children"), Input("tree", "checked"))
+    def update_output_checked(checked):
+        return str(checked)
```

Comment:
When I was playing with this feature I found the ordering of lists returned by the tree to be confusing, as it depends in odd ways on the order of operations. So although the tests as you've written them are robust for the current version of Mantine, I wouldn't be surprised if in the future they change this. Also the tests themselves would be easier to read if these fields were sorted - which as you've constructed it is also the order on screen.

So I'd suggest changing this line to:

```suggestion
        return str(sorted(checked) if checked else checked)
```
(the `if checked else checked` just lets `None` through without an error, not sure if that's necessary) and the same for `expanded` and `selected` (not `data` of course 😄) and adjusting the test assertions to match.

And perhaps worth mentioning this inconsistent order in the docs?